### PR TITLE
Dex 68123 check for participant

### PR DESCRIPTION
**Pull request by evansmith**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- This PR checks for the presence of user.participant_id before calling get_dependents due to an error that BGS throws if no participant id is present. 
- If a user does not have a participant_id and goes to view dependents an error is thrown.
- Check for participant_id before making the call to bgs. return an empty persons array instead.
- Benefits and Claims - Dependents

## Related issue(s)
[- department-of-veterans-affairs/va.gov-team#0000](https://github.com/department-of-veterans-affairs/va.gov-team/issues/56456)

## Testing done

- Updated the unit tests, confirmed behavior on the view-change-dependents/view/ page. Checked against passing in nil in the logic for the method.
- Go to /view-change-dependents/view/ either as a user with no participant id or pass in nil as participant_id before running get_dependents. Run spec tests. 
- Updated the spec tests to run against a user with no participant id. Tested in browser.

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
View Dependents page.

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by tfink419 — DISMISSED** [review]
Looks good

**Review by ryan-mcneil — DISMISSED** [review]
👍

File: spec/factories/users.rb [hunk 1]
```diff
@@ -315,6 +315,17 @@
       ssn { '796043735' }
     end
 
+    factory :evss_user_no_participant, traits: [:loa3] do
```
Comment (bosawt) [inline spec/factories/users.rb:318]: I won't block the PR on this, but I'd probably rewrite the spec so it didn't create a new factory, just doesn't seem necessary

File: spec/services/bgs/dependent_service_spec.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 RSpec.describe BGS::DependentService do
   let(:user) { FactoryBot.create(:evss_user, :loa3, birth_date:) }
+  let(:user2) { FactoryBot.create(:evss_user_no_participant, :loa3, birth_date:) }
```
Comment (bosawt) [inline spec/services/bgs/dependent_service_spec.rb:7]: If you do this then you don't need to create another factory: 
`let(:user2) { FactoryBot.create(:evss_user, :loa3, participant_id: nil, birth_date:) }`

**Review by bosawt — APPROVED** [review]
Thanks for those changes, looks good and clean!

**Review by ryan-mcneil — APPROVED** [review]
👌

**Review by tblackwe — APPROVED** [review]

**Pull request merged**